gammapy/modeling/fit.py

```diff
@@ -895,3 +899,82 @@ def _repr_html_(self):
             return self.to_html()
         except AttributeError:
             return f"<pre>{html.escape(str(self))}</pre>"
+
+
+class FitResults(collections.abc.Sequence):
+    def __init__(self, results, axis=None):
+        if axis and axis.nbin != len(results):
+            raise ValueError("Axis length does not correspond to number of results")
+
+        self.results = results
+        self.axis = axis
+
+    def __add__(self, other):
+        if isinstance(other, FitResult):
+            return FitResult([*self, other])
+        else:
+            raise TypeError(f"Invalid type: {other!r}")
+
+    def __getitem__(self, key):
+        if isinstance(key, np.ndarray) and key.dtype == bool:
+            return self.__class__(list(np.array(self.results)[key]))
+        else:
+            return self.__class__(self.results[key])
+
+    def __len__(self):
+        return len(self.results)
+
+    def success(self):
+        return [f.success for f in self.results]
+
+    def models(self):
+        return Models([model for f in self.results for model in f.models])
```

Comment:
each FitResult can have multiple models. Is it safe to put all of them in a single Models object?  It is not guaranteed that all the fitresult.models will have the same number of models either

Reply:
You are completely right, I was thinking mostly of a possible output from an estimator (which we can suppose to have the same output model "shape") but the FitResults object could be used for many other cases... I think an option ("flatten"-like) + logging if the models don't have the same length.